src/game/server.cpp

```diff
@@ -1058,29 +1060,27 @@ namespace server
             if(!map || !*map) map = choosemap(suggest, mode, muts, G(rotatemaps), true);
             else if(strchr(map, ' '))
             {
-                static string defaultmap;
-                defaultmap[0] = '\0';
-                vector<char *> maps;
+                static std::string defaultmap;
+                std::vector<std::string> maps;
                 explodelist(map, maps);
                 if(*sv_previousmaps)
                 {
-                    vector<char *> prev;
+                    std::vector<std::string> prev;
                     explodelist(sv_previousmaps, prev);
-                    loopvj(prev) loopvrev(maps) if(strcmp(prev[j], maps[i]))
+                    loopvj(prev) loopvrev(maps) if(prev[j] != maps[i])
                     {
-                        delete[] maps[i];
-                        maps.remove(i);
-                        if(maps.length() <= 1) break;
+                        maps.erase( maps.begin() + i );
+                        if(maps.size() <= 1) break;
                     }
-                    prev.deletearrays();
+                    prev.clear();
                 }
                 if(!maps.empty())
                 {
-                    int r = rnd(maps.length());
-                    copystring(defaultmap, maps[r]);
+                    int r = rnd(maps.size());
+                    defaultmap = maps[r];
```

Comment:
Given the parameter order in libc, one could expected `defaultmap` to be the target. But I'll have to double-check.

Reply:
Yeah, looks like it.